pkg/datapath/iptables/iptables.go

```diff
@@ -372,6 +372,19 @@ func newIptablesManager(p params) datapath.IptablesManager {
 	iptMgr.ip4tables = ip4tables
 	iptMgr.ip6tables = ip6tables
 
+	// IPTables manager cannot start properly if MASQ is enabled with no
+	// egressMasqeuradingInterfaces for any IPAM mode other then ClusterPool or
+	// Kubernetes.
+	//
+	// This is because Kubernete's PodCIDR is only guaranteed to be available
```

Comment:
nit: Typo in Kubernetes - but I would re-formulate this anyway:

```suggestion
	// This is because a PodCIDR is only guaranteed to be available
```

(since in cluster-pool, the Pod CIDR is not assigned by Kubernetes but by Cilium)